app/src/main/java/net/gsantner/markor/ui/NewFileDialog.java

```diff
@@ -155,12 +156,24 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                     }
 
                     appSettings.setNewFileDialogLastUsedExtension(fileExtEdit.getText().toString().trim());
-                    final File f = new File(basedir, fileNameEdit.getText().toString().trim() + fileExtEdit.getText().toString().trim());
+                    boolean isNewZimWikiFile = typeSpinner.getSelectedItemPosition() == 3;
+                    // zim uses underscores in the filename to represent spaces in the page title
+                    final String fileNameWithoutExtension = isNewZimWikiFile ? fileNameEdit.getText().toString().trim().replace(' ', '_')
+                            : fileNameEdit.getText().toString().trim();
+                    final String fileExtension = fileExtEdit.getText().toString().trim();
+                    final File f = new File(basedir, fileNameWithoutExtension + fileExtension);
                     final byte[] templateContents = getTemplateContent(templateSpinner, basedir, encryptCheckbox.isChecked());
                     shareUtil.writeFile(f, false, (arg_ok, arg_fos) -> {
                         try {
-                            if (f.exists() && f.length() < ShareUtil.MIN_OVERWRITE_LENGTH && templateContents != null) {
+                            boolean isAllowedToWrite = f.exists() && f.length() < ShareUtil.MIN_OVERWRITE_LENGTH;
+                            if (isAllowedToWrite && templateContents != null) {
                                 arg_fos.write(templateContents);
+                            } else {
+                                if (isAllowedToWrite && isNewZimWikiFile) {
+                                    String headerAndTitleContents = ZimWikiFileContentsCreator.createZimWikiHeaderAndTitleContents(fileNameWithoutExtension, new Date(), getResources().getString(R.string.created));
+                                    arg_fos.write(getBytesFromStringContents(encryptCheckbox.isChecked(), headerAndTitleContents));
+                                }
+
```

Comment:
@fredericjacob 
I'm not really fond of these changes. We already have a method that is soley there for handling templates.  
What I suggest is: 

* getTemplateContent()::::: Add case 8 (Empty Zim file). Return whatever you want there in `case 8: {`, but don't write any file. Thats job of this overall, not of the `ZimWikiFileContentsCreator`.
* typeSpinner.setOnItemSelectedListener():::::: When Zim selected, select template Empty Zim file (once). User may select a different template afterwards, thats up to the user. Like the advanced Zim template.

Reply:
I handled this.

Reply:
@gsantner I understand your point here. However, these changes also remove the feature I introduced: that typed file names for new pages are automatically adjusted so that the files are always compatible with Zim Desktop Wiki.
In Zim Desktop, the created text file for a new page with the title "My Page" automatically gets the file name "My_Page.txt". That's why I implemented the logic to replace spaces with underscores for the file name, so that the end user does not have to care about the internals of file naming in Zim. If a user types a name that doesn't conform this naming rule, compatibility with the desktop application will break.

If your okay with this, I will introduce a method that determines the file name to use (for zim wiki) instead of the plain calls of `fileNameEdit.getText()`. I think this will prevent users from quite some confusion why their notebook pages created with Markor are not shown in the desktop application.